packages/deployed-backend-client/src/deployed_backend_client.ts

```diff
@@ -82,23 +82,38 @@ export class DefaultDeployedBackendClient implements DeployedBackendClient {
     return this.buildBackendMetadata(stackName);
   };
 
-  /**
-   * Returns Amplify Sandboxes for the account and region. The number of sandboxes returned can vary
-   */
-  listSandboxes = async (
-    listSandboxesRequest?: ListSandboxesRequest
-  ): Promise<ListSandboxesResponse> => {
-    const stackMetadata: SandboxMetadata[] = [];
-    let nextToken = listSandboxesRequest?.nextToken;
+  listBackends = async (
+    listBackendsRequest?: ListBackendsRequest
+  ): Promise<ListBackendsResponse> => {
+    const stackMetadata: ListBackendsMetadata[] = [];
 
+    let nextToken = listBackendsRequest?.nextToken;
+    const deploymentType = listBackendsRequest?.deploymentType;
+    let stackStatusFilter: StackStatus[] = [];
+    if (deploymentType == 'branch') {
+      stackStatusFilter = ['DELETE_FAILED'];
+    }
```

Comment:
Why are we coupling the deployment type to a delete_failed stack status? We should probably have two filter criteria on ListBackendRequest, one for backend type and another for backend deployment status. We should use our BackendDeploymentStatus type rather than expose CFN status directly